### PR TITLE
Provide gas production, heating, damage energy data in HDF5 files

### PR DESCRIPTION
**Pull request by paulromano**
Gas production and damage energy production cross sections used to be part of our data files but were actually removed recently as a result of #1118. @Shimwell conveyed to me that these data are important for fusion applications, and it sounded like he was jumping through some hoops to get those data back in the file. This PR simply keeps them in the file as redundant reactions so that they don't contribute to anything, but a user can tally them by requesting the appropriate reaction:
- `H1-production` (MT=203)
- `H2-production` (MT=204)
- `H3-production` (MT=205)
- `He3-production` (MT=206)
- `He4-production` (MT=207)
- `damage-energy` (MT=444)

I've also included total neutron heating (MT=301) in the data file as well. A related update is that `IncidentNeutron.from_njoy` will also run `GASPR` as part of the NJOY run so that gas production will show up in the files produced that way as well.

### CODE REVIEW

**Comment by paulromano** [comment]
@liangjg I'm actually not sure about photon heating -- we'll have to look into that in a little more detail. Is that a topic you're interested in?

Two other updates I made in this PR:
- `IncidentNeutron.from_njoy` and `ThermalScattering.from_njoy` didn't work if the ENDF tape being processed had multiple materials. I've added optional arguments that allow one to pick a specific evaluation from a file in this case.
- Improved int/float handling for ENDF files. Notably, there are some files where the evaluator would leave an integer field blank (which implicitly means zero). That is now correctly handled.

File: openmc/data/njoy.py [hunk 1]
```diff
@@ -234,10 +239,15 @@ def make_ace(filename, temperatures=None, ace='ace', xsdir='xsdir', pendf=None,
         Indicating whether to Doppler broaden XS when running NJOY
     heatr : bool, optional
         Indicating whether to add heating kerma when running NJOY
+    gaspr : bool, optional
+        Indicating whether to add gas production data when running NJOY
     purr : bool, optional
         Indicating whether to add probability table when running NJOY
     acer : bool, optional
         Indicating whether to generate ACE file when running NJOY
+    evaluation : str, optional
```
Comment (liangjg) [inline openmc/data/njoy.py]: `str` -> `openmc.data.endf.Evaluation`

**Comment by liangjg** [comment]
@paulromano I just found there is a mistake I missed in the review of this PR. It is in heating number extraction from ace, which should be:
``` diff
--- a/openmc/data/neutron.py
+++ b/openmc/data/neutron.py
@@ -622,7 +622,7 @@ class IncidentNeutron(EqualityMixin):
         data.energy[strT] = energy
         total_xs = ace.xss[i + n_energy : i + 2*n_energy]
         absorption_xs = ace.xss[i + 2*n_energy : i + 3*n_energy]
-        heating_number = ace.xss[i + 3*n_energy : i + 4*n_energy]*EV_PER_MEV
+        heating_number = ace.xss[i + 4*n_energy : i + 5*n_energy]*EV_PER_MEV
```
I can fix this but the test library in ANL box also needs to be updated. So I suggest you may submit a PR for this? Thanks.

**Comment by paulromano** [comment]
@liangjg You're absolutely right -- I'll submit a PR to fix that.

**Pull request merged**